packages/extension/src/common/cache.js

```diff
@@ -7,7 +9,11 @@ export const FIRST_INSTALL_KEY = 'firstInstall';
 export const LAST_COMMENT_KEY = 'lastComment';
 
 export function setCache(key, value) {
-  return browser.storage.local.set({ [key]: value });
+  isPrivateMode().then(function(isPrivate) {
```

Comment:
You need to return the promise here.
Otherwise, it will not be usable from the outside

Reply:
What do you mean by that? Do you mean outside the if statement?